rust/src/treefile.rs

```diff
@@ -279,6 +279,22 @@ fn merge_vec_field<T>(dest: &mut Option<Vec<T>>, src: &mut Option<Vec<T>>) {
     }
 }
 
+/// Merge a map field similarly to Python's `dict.update()`. In case of
+/// duplicate keys, `dest` wins (`src` is the "included" config).
+fn merge_map_field<T>(
+    dest: &mut Option<HashMap<String, T>>,
+    src: &mut Option<HashMap<String, T>>)
+{
+    if let Some(mut srcv) = src.take() {
+        if let Some(mut destv) = dest.take() {
+            for (k, v) in destv.drain() {
+                srcv.insert(k, v);
```

Comment:
I think you are manually rolling the equivalent of https://doc.rust-lang.org/std/iter/trait.Extend.html#tymethod.extend.

Reply:
Heh, it's funny. I'm working on another rpm-ostree patch where I did find `.extend()` and use it. But somehow missed it would've been useful in this patch as well. Thanks, will fix!